core/src/parser/mod.rs

```diff
@@ -135,11 +146,118 @@ where
         }
     }
 
-    fn parse_strict(&self, file_id: FileId, lexer: lexer::Lexer) -> Result<T, ParseErrors> {
+    fn parse_strict(&self, file_id: CacheKey, lexer: lexer::Lexer) -> Result<T, ParseErrors> {
         match self.parse_tolerant(file_id, lexer) {
             Ok((t, e)) if e.no_errors() => Ok(t),
             Ok((_, e)) => Err(e),
             Err(e) => Err(e.into()),
         }
     }
 }
+
+/// Supported input formats.
+#[derive(Default, Clone, Copy, Eq, Debug, PartialEq)]
+pub enum InputFormat {
+    #[default]
+    Nickel,
+    Json,
+    Yaml,
+    Toml,
+    #[cfg(feature = "nix-experimental")]
+    Nix,
+    Raw,
+}
```

Comment:
I wonder if we should mix that with this module. In particular because it's been in the back of my head to move the (pure) Nickel parser in a separate crate, so that

1. It can be used by other crates to make tooling around Nickel without depending on the whole of nickel-lang-core
2. It's a separate compilation unit that we don't need to even look at if we didn't touch the grammar

I guess it's still doable, but we should be wary of coupling too much the "parser pure Nickel code" from "parse generic Nickel inputs including JSON", if possible.